mobile_payment/gql_mutations.py

```diff
@@ -0,0 +1,421 @@
+            #get wallet_address with existing uuid                                                                                                                                                                                                                                                                                                                                                                                                                                                                                                                                    
+            psp_address = psp.psp_account
+            psp_pin = psp.psp_pin
+            #Initiate a request for payment
+            Qcell_request = initiate_request(insuree_wallet,psp_address,amount,psp_pin,)
+            if Qcell_request['responseCode'] == '1':
+                data['transaction_id'] = Qcell_request['data']['transactionId']
+                transaction = update_or_create_transaction(data, info)
+            else:
+                raise Exception(_("Failed to initiate payment"))
+            return InitiateTransactionMutation(uuids=transaction.uuid, internal_id=response.internal_id)
+        except Exception as exc:
+            logger.exception("transaction.mutation.failed_to_create_transaction")
+            return [{
+                    'message': _("transaction.mutation.failed_to_create_transaction"),
+                    'detail': str(exc)}] 
+        
+    
+        
+class  ProcessTransactionMutation(OpenIMISMutation):
+    """
+    create a mutaion for making a trasaction
+    """
+
+    _mutation_module = "mobile_payment"
+    _mutation_class = "ProcessTransactionMutation"
+
+    class Input(TransactionBase, OpenIMISMutation.Input):
+        pass
+
+    @classmethod
+    def mutate_and_get_payload(cls, root, info, **data) -> Optional[str]:
+        response = super().mutate_and_get_payload(root, info, **data)
+        try:
+            if info.context.user is AnonymousUser:
+                raise ValidationError(_("mutation.authentication_required"))
+            if not info.context.user.has_perms(MobilePaymentConfig.gql_mutation_create_transaction_perms):
+                raise PermissionDenied(_("unathorized"))
+            client_mutation_id = data.get("client_mutation_id")
+            transaction_uuid = data['uuid']
+            transacton = Transactions.objects.get(uuid=transaction_uuid )
+            if not transacton:
+                raise PermissionDenied (_("invalid transaction_uuid"))
+            transaction_id = transacton.transaction_id
+            otp = data['otp']
+            print(f' Transaction_id = {transaction_id}')
+            print (f'otp =  {otp}')
+            Qcell_process = process_request(otp, transaction_id)
+            if Qcell_process['responseCode'] == '1':
+                data['status'] = True
+                update_or_create_transaction(data, info)
```

Comment:
I guess that means the transaction succeeded. If that's right, why the policy isn't updated as it is below in the case of Afrimoney?

Reply:
its done in the Frontend

Reply:
Could you share the frontend implementation doing that?

If I understand correctly, in the case of QMoney, if it's not correctly implemented in the Frontend we might get the wrong result (i.e. a policy not being updated accordingly). If it's correct, I think that's too brittle and should be implemented in the backend to provide consistency.

Reply:
Policy is updated  in the back-end if you check the function ```update_or_create_premium ``` it handles the update  of a paid contribution and stores the transaction

Reply:
Where and when is this function `update_or_create_premium` called?

Reply:
it is not called anywhere this is handled in the front-end of contribution after transaction payment is completed meaning status  is 1.  it pics the transaction detail and stores in in the premium table of contribution.

Reply:
Let me rephrase to make sure I understand. Do you mean that a part of the logic required to make it work is implemented in the frontend? Is that correct?

If it's the case, that's not an acceptable solution as someone working in the backend might miss that part. For cohesion, it's better to implement the whole logic in the backend.

Reply:
No the whole logic is implemented in the backend.

Reply:
Ok.